src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -416,28 +415,135 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
     else:
         raise CLIError('DNS routing verification failed')
 
+# Validate inputs for --repository and --image
+def validate_repository_and_image(cmd, registry_name, repository_name, image):
+    from .repository import get_access_credentials
+    from ._docker_utils import request_data_from_registry
+    from azure.cli.core.azclierror import ResourceNotFoundError
+
+    registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name)
+    login_server = registry.login_server.rstrip('/')
+    path = '/v2/' + repository_name + '/manifests/' + image
+
+    # Get the access credentials for the registry
+    registry_name, username, password = get_access_credentials(
+        cmd,
+        registry_name=registry_name,
+        tenant_suffix=None,
+        username=None,
+        password=None,
+        repository=repository_name,
+        permission='pull')
+    
+    try:
+        request_data_from_registry(
+            http_method='get',
+            login_server=login_server,
+            path=path,
+            username=username,
+            password=password,
+            json_payload=None,
+            file_payload=None,
+            manifest_headers={'Accept': 'application/vnd.docker.distribution.manifest.v2+json'},
+            params=image)
+    except CLIError as e:
+        # Error is due to the image tag not being found on manifest
+        if "manifest tagged by" in str(e):
```

Comment:
what if im looking by digest and the digest isn't found? Wouldn't this logic give me REPOSITORY_INPUT_ERROR_MESSAGE?